### PR TITLE
WIP more directive docs

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/MethodDirectivesExamplesSpec.scala [hunk 1]
```diff
@@ -99,4 +108,29 @@ class MethodDirectivesExamplesSpec extends RoutingSpec {
...
+          post {
+            complete("This looks like a POST request.")
+          }
+      }
+
+    // tests:
+    Get("/?method=POST") ~> route ~> check {
+      responseAs[String] shouldEqual "This looks like a POST request."
+    }
+    Post("/?method=get") ~> route ~> check {
```
Comment (patriknw) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/MethodDirectivesExamplesSpec.scala]: rather funny, and cool

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
-            BodyPart(entity2, `Content-Range`(RangeUnit.Bytes, range2) +: _) +: Seq()
-          ) ⇒ entity1.asString shouldEqual "ABC" and range1 shouldEqual ContentRange(0, 2, 8) and
-          entity2.asString shouldEqual "GH" and range2 shouldEqual ContentRange(6, 7, 8)
-      }
-    }
-  }
-}
-*/ 
\ No newline at end of file
+///*
```
Comment (patriknw) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: FIXME ?

Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: Working on it right now actually :)

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RespondWithDirectivesExamplesSpec.scala [hunk 3]
```diff
@@ -80,68 +80,4 @@ class RespondWithDirectivesExamplesSpec extends RoutingSpec {
...
-  //    }
-  //
-  //    Get("/bar") ~> route ~> check {
-  //      headers.filter(_.is("funky-muppet")) shouldEqual List(RawHeader("Funky-Muppet", "kermit"))
-  //      responseAs[String] shouldEqual "beep"
-  //    }
-  //  }
-
-  // FIXME https://github.com/akka/akka/issues/18626
-  //  "respondWithStatus-examples" in {
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RespondWithDirectivesExamplesSpec.scala]: these directives are not ported as deemed as anti-patterns, documented below and will be in  the full blown migration guide

**Comment by patriknw** [comment]
LGTM

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala (hunk 2 above)
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/directives/RangeDirectivesExamplesSpec.scala]: Made it a ticket https://github.com/akka/akka/issues/18706

**Comment by ktoso** [comment]
Sigh, can't figure out why pdf is failing :(

**Comment by ktoso** [comment]
Fixed the PDF issue, should pass now; Will merge.

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst [hunk 4]
```diff
@@ -3,8 +3,6 @@
 formFields
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: cc @jrudolph @sirthias can you please have a look at this file, in respect to things mentioned in https://github.com/akka/akka/issues/17938 ? Did I miss something?

**Comment by ktoso** [comment]
Merging to get it in, however would like feedback on the `formFields` for example // cc @sirthias @jrudolph thanks!

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka/issues/18496

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst [hunk 5]
```diff
@@ -61,14 +60,14 @@ Unmarshalling
 
 Data POSTed from `HTML forms`_ is either of type ``application/x-www-form-urlencoded`` or of type
 ``multipart/form-data``. The value of an url-encoded field is a ``String`` while the value of a
-``multipart/form-data``-encoded field is a "body part" containing an entity. This means that different kind of deserializers are needed depending
+``multipart/form-data``-encoded field is a "body part" containing an entity. This means that different kind of unmarshallerss are needed depending
```
Comment (jrudolph) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: s/unmarshallerss/unmarshallers/

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst [hunk 6]
```diff
@@ -36,17 +35,17 @@ as required, optional, or repeated, or to filter requests where a form field has
 ``"color" ! "blue"``
     require value of field "color" to be ``"blue"`` and extract nothing
 ``"amount".as[Int]``
-    extract value of field "amount" as ``Int``, you need a matching implicit ``Deserializer`` in scope for that to work
+    extract value of field "amount" as ``Int``, you need a matching implicit ``Unmarshaller`` in scope for that to work
     (see also :ref:`http-unmarshalling-scala`)
 ``"amount".as(deserializer)``
```
Comment (sirthias) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: s/deserializer/unmarshaller/

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst [hunk 7]
```diff
@@ -36,17 +35,17 @@ as required, optional, or repeated, or to filter requests where a form field has
...
-    extract value of field "amount" with an explicit ``Deserializer``
+    extract value of field "amount" with an explicit ``Unmarshaller``
 ``"distance".*``
     extract multiple occurrences of field "distance" as ``Iterable[String]``
 ``"distance".as[Int].*``
-    extract multiple occurrences of field "distance" as ``Iterable[Int]``, you need a matching implicit ``Deserializer`` in scope for that to work
+    extract multiple occurrences of field "distance" as ``Iterable[Int]``, you need a matching implicit ``Unmarshaller`` in scope for that to work
     (see also :ref:`http-unmarshalling-scala`)
 ``"distance".as(deserializer).*``
```
Comment (sirthias) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: s/deserializer/unmarshaller/

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst (hunk 4 above)
Comment (sirthias) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: I'd also remove the sentence referencing `anyParams`.

Comment (sirthias) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: And: In the light of #17938 we should maybe add a sentence explaining that there is nothing in this directive which filters for POST requests or any other request method type. It only concerns itself with providing convenient deserialization access to the request entity.

Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: good catch, thanks!

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst (hunk 7 above)
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: thanks :)

**Pull request merged**